src/segment-loader.js

```diff
@@ -1078,20 +1109,23 @@ export default class SegmentLoader extends videojs.EventTarget {
           this.activeInitSegmentId_ !== initId) {
         const initSegment = this.initSegment(segment.map);
 
-        this.sourceUpdater_.appendBuffer(initSegment.bytes, () => {
+        this.sourceUpdater_.appendBuffer(initSegment.bytes, null, () => {
           this.activeInitSegmentId_ = initId;
         });
       }
     }
 
     segmentInfo.byteLength = segmentInfo.bytes.byteLength;
+
+    let segmentDuration;
+
     if (typeof segment.start === 'number' && typeof segment.end === 'number') {
-      this.mediaSecondsLoaded += segment.end - segment.start;
+      this.mediaSecondsLoaded += segmentDuration = segment.end - segment.start;
```

Comment:
Makes sense, but is a bit hard to read. Maybe the conditionals just set the value of `segmentDuration` (or a ternary is used), then `this.mediaSecondsLoaded += segmentDuration` exists after?